### PR TITLE
Added a Riak backend to CCDB

### PR DESCRIPTION
**Pull request by kouzinopoulos**

### CODE REVIEW

File: CCDB/src/ConditionsMQServer.cxx [hunk 1]
```diff
@@ -9,23 +9,30 @@
  * ConditionsMQServer.cxx
```
Comment (matthiasrichter) [inline CCDB/src/ConditionsMQServer.cxx]: I still need to understand how much the CnditionsMQServer is tailored to the old OCDB case. The general server should not depend on these things. If the current implementation is a specialization for the OCDB case we have to think about a different name.

File: CCDB/src/runConditionsClient.cxx [hunk 2]
```diff
@@ -1,80 +1,84 @@
...
 #include "FairMQLogger.h"
 #include "FairMQParser.h"
 #include "FairMQProgOptions.h"
 
-#include "CCDB/ConditionsMQClient.h"
+#ifdef NANOMSG
+#include "FairMQTransportFactoryNN.h"
+#else
+#include "FairMQTransportFactoryZMQ.h"
+#endif
```
Comment (matthiasrichter) [inline CCDB/src/runConditionsClient.cxx]: The explicit code choosing the transport layer is not necessary any more. It is handled inside the FairMQ boiler plate code. This code shows many signs of copying from another outdated code file. FairMQ functionality and boiler plate code has improved a lot, so this needs a rewrite.

File: cmake/O2Dependencies.cmake [hunk 3]
```diff
@@ -27,6 +27,7 @@ find_package(ZeroMQ)
 find_package(AliRoot)
 find_package(FairRoot REQUIRED)
 find_package(FairMQ REQUIRED)
+find_package(Protobuf REQUIRED)
```
Comment (matthiasrichter) [inline cmake/O2Dependencies.cmake]: Is it necessary to make protobuf a required dependency? Or can the compilation be conditional? Lets consider protobuf as one way of serialization and not bind the code completely to it.

File: CCDB/src/runConditionsClient.cxx (hunk 2 above)
Comment (kouzinopoulos) [inline CCDB/src/runConditionsClient.cxx]: So SetTransport(transportFactory) is not needed at all? When i try to remove it i get the following error:

Transport not initialized. Did you call SetTransport()?

Also, i grepped through the source code and it seems that the same code is used by other devices as well so probably these have to be updated too

Comment (rbx) [inline CCDB/src/runConditionsClient.cxx]: If you use the new approach (as in FairMQ examples & flp2epn examples), you don't need these lines, the transport is chosen via a cmd option `--transport` (default it zeromq), which is handled internally.

But yeah, besides the flp2epn examples, there are likely other outdated devices in the repository.

**Comment by MohammadAlTurany** [comment]
Hi Charis,
I am not able to compile the code, I get the error:

`AliceO2/CCDB/src/ObjectHandler.cxx:3:10: fatal error: 'AliCDBEntry.h' file not found  #include “AliCDBEntry.h"`

Does it depend on AliRoot? 
best,
Mohammad

**Comment by kouzinopoulos** [comment]
Hi Mohammad, 

The code depends on AliRoot to load and stream OCDB objects to a TBufferFile using AliCDBEntry. These objects we then compress/serialize and PUT/GET to Riak.

**Pull request merged**